### PR TITLE
ILE-3 Client

### PR DESCRIPTION
**Pull request by dylanmckk**

### CODE REVIEW

File: src/main/java/org/example/controllers/ClientController.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        }
+
+    }
+
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createClient(
+            final ClientRequest clientRequest) {
+        try {
+            return Response.ok()
```
Comment (lukemKainos) [inline src/main/java/org/example/controllers/ClientController.java:69]: this could be a 201 instead as its created?

File: src/main/java/org/example/daos/DatabaseConnector.java [hunk 2]
```diff
@@ -26,8 +26,8 @@ public static Connection getConnection() throws SQLException {
                     || name == null) {
                 throw new IllegalArgumentException(
                         "Add the following properties to env vars: "
-                                +
-                                "DB_USERNAME, DB_PASSWORD, DB_HOST and DB_NAME");
+                        +
+                        "DB_USERNAME, DB_PASSWORD, DB_HOST and DB_NAME");
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/DatabaseConnector.java]: why is the formatting weird here

File: src/main/java/org/example/daos/SalesEmployeeDao.java [hunk 3]
```diff
@@ -163,4 +146,24 @@ public void deleteSalesEmployee(final int id) throws SQLException {
...
     }
+
+    private enum Indexes {
+        firstName(1),
+        lastName(2),
+        salary(3),
+        bankAccount(4),
+        commissionRate(5),
+        nationalInsuranceNumber(6),
+        updateId(7);
```
Comment (lukemKainos) [inline src/main/java/org/example/daos/SalesEmployeeDao.java]: enum should be in accordance with naming conventions i.e. FIRST_NAME

File: src/main/java/org/example/models/Client.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package org.example.models;
+
+public class Client {
+    private int clientID;
```
Comment (lukemKainos) [inline src/main/java/org/example/models/Client.java]: really should technically be Id

File: src/main/java/org/example/models/Client.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        this.name = name;
+    }
+
+    public int getClientID() {
+        return clientID;
+    }
+
+    public void setClientID(final int clientID) {
+        this.clientID = clientID;
+    }
```
Comment (lukemKainos) [inline src/main/java/org/example/models/Client.java]: is this being used?

File: src/main/java/org/example/responses/ClientResponse.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    public ClientResponse(final int clientId, final String name) {
+        this.clientId = clientId;
+        this.name = name;
+    }
+
+    public int getClientId() {
+        return clientId;
+    }
+
+    public void setClientId(final int clientId) {
```
Comment (lukemKainos) [inline src/main/java/org/example/responses/ClientResponse.java]: why is this needed?

File: src/main/java/org/example/validators/ClientValidator.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        if (clientRequest.getName().length() > 50) {
+            throw new InvalidException(Entity.CLIENT,
+                    "Name greater than 50 characters.");
+        }
+
+
+        if (clientRequest.getName().isEmpty() ||
+                clientRequest.getName() == null) {
+            throw new InvalidException(Entity.CLIENT,
+                    "Name greater is blank. Please enter a name.");
```
Comment (lukemKainos) [inline src/main/java/org/example/validators/ClientValidator.java]: what?

**Review by lukemKainos — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/daos/ClientDao.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        List<Client> clients = new ArrayList<>();
+        try (Connection connection = DatabaseConnector.getConnection()) {
+            Statement statement = connection.createStatement();
+
+            ResultSet resultSet =
+                    statement.executeQuery("SELECT * FROM client;");
+
+            while (resultSet.next()) {
+                Client client = new Client(
+                        resultSet.getInt("clientID"),
```
Comment (dana-kainos) [inline src/main/java/org/example/daos/ClientDao.java]: should this be clientId?

File: src/main/java/org/example/daos/ClientDao.java [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+            String query =
+                    "SELECT * FROM client WHERE clientId = ?";
+            PreparedStatement statement = connection.prepareStatement(query);
+
+            statement.setInt(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                return new Client(
+                        resultSet.getInt("clientID"),
```
Comment (dana-kainos) [inline src/main/java/org/example/daos/ClientDao.java]: see above

**Review by dana-kainos — CHANGES_REQUESTED** [review]

**Review by dana-kainos — APPROVED** [review]

File: src/main/java/org/example/controllers/ClientController.java (hunk 1 above)
Comment (lukemKainos) [reply src/main/java/org/example/controllers/ClientController.java:69]: this isnt a 201

Comment (dylanmckk) [reply src/main/java/org/example/controllers/ClientController.java:69]: If you look below, ive put '.status(Response.Status.CREATED)', which returns 201?

Comment (dylanmckk) [reply src/main/java/org/example/controllers/ClientController.java:69]: Line 70

**Review by lukemKainos — APPROVED** [review]
